src/Microsoft.AspNet.Server.Kestrel/KestrelServer.cs

```diff
@@ -75,35 +75,14 @@ public void Start(RequestDelegate requestDelegate)
                 _disposables.Push(engine);
                 _disposables.Push(dateHeaderValueManager);
 
-                // Actual core count would be a better number 
-                // rather than logical cores which includes hyper-threaded cores.
-                // Divide by 2 for hyper-threading, and good defaults (still need threads to do webserving).
-                // Can be user overriden using IKestrelServerInformation.ThreadCount
-                var threadCount = Environment.ProcessorCount >> 1;
+                var threadCount = information.ThreadCount;
 
-                if (threadCount < 1)
+                if (threadCount < 0)
```

Comment:
This PR looks good, sorry for the delayed response. The only thing I would change is making this `threadCount <= 0` and the exception message ""ThreadCount must be positive."